updater/sis-updater-worker/src/updater/updateStudents/attainments.js

```diff
@@ -135,53 +135,44 @@ const updateAttainments = async (
       }
 
       if (!customAttainmentTypes.includes(att.type)) return att
-      const courseUnits = courses.filter(c => c.code === attIdToCourseCode[att.id])
-
-      const isAfterStartAndBeforeEnd = (startDate, endDate, date) => {
-        const dateToCompare = new Date(date)
-        return new Date(startDate) <= dateToCompare && (!endDate || dateToCompare < new Date(endDate))
-      }
-
-      let courseUnit = courseUnits.find(({ validity_period }) =>
-        isAfterStartAndBeforeEnd(validity_period.startDate, validity_period.endDate, att.attainment_date)
-      )
-
-      // Sometimes registrations are fakd, see attainment hy-opinto-141561630.
-      // The attainmentdate is outside of all courses, yet should be mapped.
-      // Try to catch suitable courseUnit for this purpose
-      if (!courseUnit) {
-        courseUnit = courseUnits.find(({ validity_period }) =>
-          isAfterStartAndBeforeEnd(validity_period.startDate, validity_period.endDate, att.registration_date)
-        )
-      }
+      let courseUnit = courses.find(c => c.id === att.course_unit_id)
 
       // If there's no suitable courseunit, there isn't courseunit available at all.
       // --> Course should be created, if it doesn't exist in sis db
       if (!courseUnit) {
         const parsedCourseCode = attIdToCourseCode[att.id]
-        // see if course exists
-        const course = await Course.findOne({ where: { code: parsedCourseCode } })
-
-        // If course doesn't exist, create it
-        if (!course) {
-          coursesToBeCreated.set(parsedCourseCode, {
-            id: parsedCourseCode,
-            name: att.name,
-            code: parsedCourseCode,
-            coursetypecode: att.study_level_urn,
-            course_unit_type: att.course_unit_type_urn,
-          })
+        courseUnit = {
+          id: att.id,
+          name: att.name,
+          code: att.code,
+          latest_instance_date: att.attainment_date,
+          is_study_module: !(
```

Comment:
This should probably be
```js
is_study_module: [
  'StudyModuleAttainment',
  'CustomStudyModuleAttainment',
  'DegreeProgrammeAttainment',
  'CustomDegreeProgrammeAttainment',
].includes(att.type),
```
or better yet, add the missing module types to https://github.com/UniversityOfHelsinkiCS/oodikone/blob/548f3fc622395e143757238e41de52a0c7c3c3d4/updater/sis-updater-worker/src/updater/mapper.js#L34-L41

and export the `isModule` function. The missing custom types should also be added to `customAttainmentTypes`.

Also, there are only `ModuleAttainments` and `CustomModuleAttainments` in our db, has it been checked that `StudyModuleAttainments` and `CustomStudyModuleAttainments` also exist?

Reply:
Yea checked from ori swagger and there is not `StudyModuleAttainments` and `CustomStudyModuleAttainments`

Reply:
Should be ok now